core/primitives/tipset/tipset.hpp

```diff
@@ -29,19 +29,30 @@ namespace fc::primitives::tipset {
 OUTCOME_HPP_DECLARE_ERROR(fc::primitives::tipset, TipsetError);
 
 namespace fc::primitives::tipset {
+  using block::Address;
   using block::BeaconEntry;
   using block::BlockHeader;
   using common::Hash256;
   using crypto::randomness::DomainSeparationTag;
   using crypto::randomness::Randomness;
+  using vm::message::SignedMessage;
+  using vm::message::UnsignedMessage;
 
   struct MessageVisitor {
-    using Visitor =
-        std::function<outcome::result<void>(size_t, bool bls, const CID &)>;
+    using Visitor = std::function<outcome::result<void>(size_t,
+                                                        bool bls,
+                                                        const CID &,
+                                                        SignedMessage *smsg,
+                                                        UnsignedMessage *msg)>;
```

Comment:
Why 2 pointers?
It could be optional to not use nullptr,
Or variant because we can get UnsignedMessage from SignedMessage (msg == msg.message) as I understand.

Reply:
`smsg == nullptr` when message is bls.
May pass `SignedMessage &`, but signature will be undefined and should not be used.

Reply:
Variant?

Reply:
We need `UnsignedMessage` for both bls and secp.
If we use `boost::variant<SignedMessage, UnsignedMessage>`, then we need one more variable to store `UnsignedMessage` anyway.

Reply:
ok